### PR TITLE
Update mongodb atlas plugin version

### PR DESCRIPTION
**Pull request by fairclothjm**
## Description
Update go.mod to use the newest version of the Vault MongoDB Atlas plugin which adds the ability to customize username generation for dynamic users in MongoDB Atlas.

## Related issues
- Plugin update PR: https://github.com/hashicorp/vault-plugin-database-mongodbatlas/pull/26
- Vault Docs update PR: https://github.com/hashicorp/vault/pull/11943

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -105,7 +105,7 @@ require (
 	github.com/hashicorp/vault-plugin-secrets-openldap v0.4.0
 	github.com/hashicorp/vault-plugin-secrets-terraform v0.1.0
 	github.com/hashicorp/vault/api v1.0.5-0.20210210214158-405eced08457
-	github.com/hashicorp/vault/sdk v0.1.14-0.20210505171055-299f311fa707
+	github.com/hashicorp/vault/sdk v0.2.0
```
Comment (pcman312) [inline go.mod:108]: @calvn is this correct? Isn't the commit this is replacing later than 0.2.0 of the SDK?

Comment (calvn) [reply go.mod:108]: Yeah, can this be reverted?

Comment (calvn) [reply go.mod:108]: Actually, this may not matter because of the replace directive at the very top.

**Review by pcman312 — APPROVED** [review]

**Review by calvn — APPROVED** [review]

**Comment by calvn** [comment]
Related to #11943

**Comment by calvn** [comment]
nit: can you move the changelog that's in #11943 over to this PR?

**Pull request merged**